src/components/RadioButton/RadioButton.scss

```diff
@@ -3,45 +3,94 @@
 $block: '.#{variables.$ns}radio-button';
 
 #{$block} {
+    --_--border-width: 1px;
+    --_--transition-time: 0.15s;
+
     box-sizing: border-box;
     display: inline-flex;
     flex-direction: row;
     font-family: var(--g-text-body-font-family);
     font-weight: var(--g-text-body-font-weight);
-    border-radius: var(--_--border-radius);
-    background-color: var(--g-color-base-generic);
     position: relative;
 
-    --_--border-radius-inner: calc(var(--_--border-radius) - 3px);
-
-    &__plate {
-        position: absolute;
-        inset-block: 0;
-        transition:
-            left 0.2s,
-            width 0.2s;
-
-        &[hidden] {
-            display: none;
-        }
-    }
-
     &__option {
+        position: relative;
         flex: 1 1 auto;
         user-select: none;
         font-size: var(--g-text-body-1-font-size);
         text-align: center;
-        border-radius: var(--_--border-radius-inner);
+
         cursor: pointer;
         transform: scale(1);
-        transition: color 0.15s linear;
+        transition: color var(--_--transition-time) linear;
 
-        &-outline {
+        &::before {
```

Comment:
What is the reason to use pseudo-elements? Why just don't change border and of element itself?